builtin/providers/google/resource_compute_instance_group.go

```diff
@@ -0,0 +1,301 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceComputeInstanceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeInstanceGroupCreate,
+		Read:   resourceComputeInstanceGroupRead,
+		Update: resourceComputeInstanceGroupUpdate,
+		Delete: resourceComputeInstanceGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"fingerprint": &schema.Schema{
```

Comment:
Well, the merge is really only interesting for the `instances` list, which your read doesn't update. What the other GCE resources do in a similar case is in the `Read` (which is automatically called before `Update` by Terraform) they only record changes to instances that were initially added to the local TF file. For example, you specify instances `A, B, C`, someone else adds `D` and deletes `A`, and then `Read` records `B, C` for the local state. The rational is, if you recorded `D` as existing in the local state, then the local state would show an attribute existing that wasn't declared in the local TF file, which would then be deleted in the next `Update`. Recording the deletion of `A` is important, as it keeps the user state in sync with the server state. The last part of the merge is then between the new and old state in `Update`.